modules/wallet/seed.go

```diff
@@ -120,6 +137,11 @@ func (w *Wallet) nextPrimarySeedAddress(tx *bolt.Tx) (types.UnlockConditions, er
 	// conditions.
 	spendableKey := generateSpendableKey(w.primarySeed, progress)
 	w.keys[spendableKey.UnlockConditions.UnlockHash()] = spendableKey
+
+	// Remove new key from the future keys and update them according to new progress
+	delete(w.futureKeys, spendableKey.UnlockConditions.UnlockHash())
+	w.updateFutureKeys(progress + 1)
```

Comment:
since you operate directly on `w.futureKeys` for the deletion, I think it's cleaner to do the same for the insertion:
```go
nextKey := generateSpendableKey(w.primarySeed, progress+1)
w.futureKeys[nextKey.UnlockConditions.UnlockHash()] = progress + 1
```